news/2025/2025-01-27-performance-points-star-rating-updates.md

```diff
@@ -0,0 +1,224 @@
+- A [change](https://github.com/ppy/osu/pull/31449) proposed by [StanR](https://osu.ppy.sh/users/7217455) to simplify the angle bonus formulas in aim calculations
+- A [change](https://github.com/ppy/osu/pull/21211) proposed by [tsunyoku](https://osu.ppy.sh/users/11315329) to implement basic difficulty and performance calculations for the Autopilot mod
+
+## osu!taiko
+
+In order to aid understanding of the changes to osu!taiko, these are the skills in difficulty calculation which will be referenced throughout:
+
+- **Stamina**: the speed at which you hit notes, based on an assumed finger count of 2 per colour
+- **Colour**: the frequency or variability of which the beatmap changes between a don or kat
+- **Rhythm**: the complexity of the beatmap's rhythm in relation to notes' independent rhythm ratios
+
+Throughout these changes, we will be using the following terminology:
+
+- **Monos** refer to repeated notes of the same colour
+- **Intervals** refer to the snap interval of notes (such as 1/1, 1/4 and 1/8)
+
+### osu!taiko committee
+
+Since the last performance point post, there are some new faces in the osu!taiko pp committee:
+
+- [YaniFR](https://osu.ppy.sh/users/11260982)
+- [BabySnakes](https://osu.ppy.sh/users/4669728)
+- [rloseise](https://osu.ppy.sh/users/6793778)
+
+### Rhythm rewrite
+
+A [set](https://github.com/ppy/osu/pull/31284) [of](https://github.com/ppy/osu/pull/31339) [changes](https://github.com/ppy/osu/pull/31573) proposed by [ltca](https://osu.ppy.sh/users/11475208) (with contributions from [rloseise](https://osu.ppy.sh/users/6793778) and [YaniFR](https://osu.ppy.sh/users/11260982)) has been created in order to rewrite the rhythm skill to address shortcomings of the previous implementation.
+
+The new rhythm skill works by creating ratios to assess difficulty. These ratios are dictated by the time between note changes — simple rhythms such as 1/2 will receive less of a bonus than more complicated rhythms such as 1/6 and 1/8, especially when frequently placed. In order to ensure this does not overly award small sections of difficulty, the frequency of these ratios are assessed to decide how difficult the ratio changes are.
+
+To handle patterns where the time between notes doesn't change much, we group evenly spaced notes into a pattern. These are often the "streams" you see in fast beatmaps. If the timing between notes is consistent, those notes go into the same group. 
+
+When the timing changes, the system shifts notes into the group with the shorter interval:
+
+- In **●● ●●●**, the third note joins the later group (**●●●**) because it fits better with the shorter timing.
+- In **●●● ●●**, the third note sticks with the earlier group (**●●●**) for the same reason.
+
+This prevents evenly spaced notes from being unfairly treated as difficult, whilst also helping to measure how rhythm and colour difficulty interact with eachother.
```

Comment:
this is a rather weird explanation.. what is the significance of the 3rd note in these patterns anyway? this is just explaining that the notes are grouped, which i guess didn't happen before

```suggestion
Similarly, when the time between notes don't change much, evenly spaced notes are treated as separate patterns (i.e. **●● ●●●** is treated as two groups). For "streams" in fast beatmaps, these groups can become large.

This prevents evenly spaced notes from being unfairly treated as difficult, while also helping to measure how rhythm and colour difficulty interact with each other.
```